### ISSUE TITLE
Something wrong 100% accuracy when network not learning

### ISSUE THREAD

**Issue by mrgloom** [NVIDIA/DIGITS#1882]
![1](https://user-images.githubusercontent.com/4003908/32845234-49e27ce6-ca35-11e7-8cdd-5af070969f5e.PNG)

Seems log show 100% accuracy, but network not learning anything, looks like bug.

```
I1114 00:15:44.863437 35326 solver.cpp:397]     Test net output #0: accuracy = 0.5
I1114 00:15:44.863507 35326 solver.cpp:397]     Test net output #1: loss = 0.696598 (* 1 = 0.696598 loss)

I1114 00:22:27.343005 35326 solver.cpp:397]     Test net output #0: accuracy = 1
I1114 00:22:27.343061 35326 solver.cpp:397]     Test net output #1: loss = 87.3414 (* 1 = 87.3414 loss)
```
